File: pertemuan10/PR/PR10_2473021_PY/A_SimpleRPG.py
"""
Pertemuan 10
[PR]  program simple RPG

@author 2473021-Febrianus Leona Putra
@version 02 Desember 2024

"""
import random

subjek = [
    ["Pahlawan", 100],
    ["Monster", 90]
]
damage_pahlawan = []
damage_monster = []

while subjek[0][1] > 0 and subjek[1][1] > 0:
    
    damage = random.randint(5, 24)
    while damage in damage_pahlawan:
        damage = random.randint(5, 24)
    damage_pahlawan.append(damage)
    subjek[1][1] -= damage
    print(f"{subjek[0][0]} menyerang {subjek[1][0]} dan menyebabkan {damage} damage.")
    print(f"HP {subjek[0][0]}: {subjek[0][1]}")
    print(f"HP {subjek[1][0]}: {max(0, subjek[1][1])}\n")    
    
    if subjek[1][1] <= 0:
        break
    
    damage = random.randint(5, 24)
    while damage in damage_monster:
        damage = random.randint(5, 24)
    damage_monster.append(damage)
    subjek[0][1] -= damage
    print(f"{subjek[1][0]} menyerang {subjek[0][0]} dan menyebabkan {damage} damage.")
    print(f"HP {subjek[0][0]}: {max(0, subjek[0][1])}")
    print(f"HP {subjek[1][0]}: {subjek[1][1]}\n")

if subjek[0][1] > 0:
    print("Pahlawan menang!")
else:
    print("Pahlawan kalah!")

File: pertemuan10/PR/PR10_2473021_JAVA/A_SimpleRPG.java

/**
 * Pertemuan 10
 * [PR]  program simple RPG
 *
 * @author 2473021-Febrianus Leona Putra
 * @version 02 Desember 2024
 */
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class A_SimpleRPG {

    public static void main(String[] args) {

        String[][] subjek = {{"Pahlawan", "100"}, {"Monster", "90"}};
        List<Integer> damagePahlawan = new ArrayList<>();
        List<Integer> damageMonster = new ArrayList<>();
        Random random = new Random();
        int hpPahlawan = Integer.parseInt(subjek[0][1]);
        int hpMonster = Integer.parseInt(subjek[1][1]);

        while (hpPahlawan > 0 && hpMonster > 0) {
            int damage = random.nextInt(20) + 5;
            while (damagePahlawan.contains(damage)) {
                damage = random.nextInt(20) + 5;
            }
            damagePahlawan.add(damage);
            hpMonster -= damage;
            System.out.println(subjek[0][0] + " menyerang " + subjek[1][0] + " dan menyebabkan " + damage + " damage.");
            System.out.println("HP " + subjek[0][0] + ": " + hpPahlawan);
            System.out.println("HP " + subjek[1][0] + ": " + Math.max(0, hpMonster) + "\n");

            if (hpMonster <= 0) {
                break;
            }

            damage = random.nextInt(20) + 5;
            while (damageMonster.contains(damage)) {
                damage = random.nextInt(20) + 5;
            }
            damageMonster.add(damage);
            hpPahlawan -= damage;
            System.out.println(subjek[1][0] + " menyerang " + subjek[0][0] + " dan menyebabkan " + damage + " damage.");
            System.out.println("HP " + subjek[0][0] + ": " + Math.max(0, hpPahlawan));
            System.out.println("HP " + subjek[1][0] + ": " + hpMonster + "\n");
        }

        if (hpPahlawan > 0) {
            System.out.println("Pahlawan menang!");
        } else {
            System.out.println("Pahlawan kalah!");
        }
    }
}
